rust_src/src/lisp.rs

```diff
@@ -924,22 +924,25 @@ impl LispObject {
     #[inline]
     pub fn as_string(self) -> Option<LispStringRef> {
         if self.is_string() {
-            Some(LispStringRef::new(
-                unsafe { mem::transmute(self.get_untaggedptr()) },
-            ))
+            Some(unsafe { self.as_string_unchecked() })
         } else {
             None
         }
     }
 
     #[inline]
     pub fn as_string_or_error(self) -> LispStringRef {
-        if self.is_string() {
```

Comment:
Why not emulate `as_window`

      self.as_string()
            .unwrap_or_else(|| wrong_type!(Qstringp, self))

Reply:
Sounds good, will update